core/textile/model/mirror_file.go

```diff
@@ -59,6 +60,43 @@ func (m *model) CreateMirrorBucket(ctx context.Context, bucketSlug string, mirro
 	return bucket, nil
 }
 
+func (m *model) FindMirrorFileByPaths(ctx context.Context, paths []string) (map[string]*MirrorFileSchema, error) {
+	metaCtx, dbID, err := m.initMirrorFileModel(ctx)
+	if err != nil || dbID == nil {
+		return nil, err
+	}
+
+	var qry *db.Query
+	for i, path := range paths {
+		if i == 0 {
+			qry = db.Where("path").Eq(filepath.Clean(path))
+		} else {
+			qry = qry.Or(db.Where("path").Eq(filepath.Clean(path)))
+		}
+	}
+
+	rawMirrorFiles, err := m.threads.Find(metaCtx, *dbID, mirrorFileModelName, qry, &MirrorFileSchema{})
+	if err != nil {
+		return nil, err
+	}
+
+	if rawMirrorFiles == nil {
+		return nil, nil
+	}
+
+	mirror_files := rawMirrorFiles.([]*MirrorFileSchema)
+	if len(mirror_files) == 0 {
+		return nil, nil
```

Comment:
Maybe these should return a "path not found" error.

Reply:
The thing is if this returns an error instead, it will break the list dirs call and I was thinking it should still return what it can with list dirs so the files at least show up even if the mirror info wasn't able to be fetched.  Also the similar function `FindMirrorFileByPathAndBucketSlug` does the same so it stays consistent with that. In that case we check that before creating and if its empty we go ahead and create a mirror file (if we return error there it would error our add files).  I guess the other option is to return `pathNotFound` in both cases but handle the error in the calling code but figured it's okay for a `find` call to return 0 results but not be an error state similar to the underlying threads Find call.

Reply:
@dmerrill6 @jsonsivar 

Maybe we can revisit these later and add `skipOnError` or similar flag?

Reply:
@maurycy 
Don't really like how optional args are handled in Go so feels a bit overkill to change the method return given that flag. In that case we could do some nomenclature agreement like `FetchX` errors if X is not found and `FindX` returns nil if not found. 

@jsonsivar 
I was thinking returning `nil, nil` with a map in the response could lead to errors if not handled on the caller side (e.g. if the caller does `if err == nil then doSomething(map[key])` would panic). But yea it's the same issue for the other Find method so I think we can revisit later and update the callers to do something like `if err.name == "ErrNotFound" then continue`.

Reply:
That's a good point. I like aligning on that nomenclature as well. Will create a card to clean that up and put it in our beta board.